### ISSUE TITLE
(Feature request) replace web login page with Basic Auth

### ISSUE THREAD

**Issue by d03j** [TriliumNext/trilium#4474]
### Describe feature

Replace web login page with Basic Auth pop up. This way, until the user is logged in they see nothing identifying what service is running. No big "trilium", logo of favico on the page.

**Comment by rauenzi** [comment]
I think _if_ this were to be added it would be better as an optional alternative you have to enable purposefully. The current method is much more user friendly.

**Comment by deajan** [comment]
@d03j Why not put a proxy in front of trilium server which can handle the basic auth ?

**Comment by d03j** [comment]
> @d03j Why not put a proxy in front of trilium server which can handle the basic auth ?

How would that work? Would that not break my laptop client's sync with the server? Would I have to enter my passwor 3 times to get into trilium web (Basic Auth, current server auth, unlocking nodes)?

**Comment by sigaloid** [comment]
No, you can disable the server auth in the settings file. Then if you have auth in the web proxy, you'll get it through there instead of Trilium.

And yes, there'll still be a PW for protected notes - the way I see it is, one password is to get into the house, one password is to get into the safe - separate section for extra-secure notes.

**Comment by d03j** [comment]
> No, you can disable the server auth in the settings file. Then if you have auth in the web proxy, you'll get it through there instead of Trilium.

ok. I'll try that.

**Comment by d03j** [comment]
> > @d03j Why not put a proxy in front of trilium server which can handle the basic auth ?
> 
> How would that work? Would that not break my laptop client's sync with the server? Would I have to enter my password 3 times to get into trilium web (Basic Auth, current server auth, unlocking nodes)?

I enabled Basic Auth on the reverse proxy as @deajan suggested but that breaks my laptop instance syncing with the server.

**Comment by d03j** [comment]
> Yeah, you'd need to give an exception to the sync endpoints.

I'm happy to test this. Where can I find the endpoints and how do you authenticate access to the currently?

**Comment by zadam** [comment]
`/api/sync/*` and `/api/login/sync`

Sync is a background process which has no knowledge of a password.

The authentication is based on a secret identity of the document. The process is verifying that the same document database sits on both sides of the sync. The secret identity is HMAC'd with a timestamp (so the identity itself is never given away).

More info in the source code: https://github.com/zadam/trilium/blob/master/src/routes/api/login.js

**Comment by d03j** [comment]
> `/api/sync/*` and `/api/login/sync`
> 

Exceptions for `/api/sync/*`, `/api/login/sync` and `/api/setup/status` worked for me.

I would still love to have the option to enable Basic Auth on the webserver and API though. :)

**Comment by foozzi** [comment]
@zadam, could you please add the config above as an example configuration for **Basic Auth** to the [wiki](https://github.com/zadam/trilium/wiki/Server-installation)?

**Comment by deajan** [comment]
Just as a side note, one could add any kind of authentication, or even TOTP / MFA in front of Trilium server by proxying it with a tool like https://github.com/authelia/authelia (or keycloack or authentik, whatever floats your boat).

**Comment by d03j** [comment]
Yes and, if I wasn't using podman rootless containers, I could create a network just for trilium and traefik only and not expose trilium's port to the host, disable trilium's server auth and have traefik be the only entry point to trilium.

Unfortunately the only way I found to get podman rootless containers talking to each other is by exposing ports to the host and use trefik's file provider instead of the docker provider. So, if I were to disable Trilium's server auth and got UFW to block anyone on my lan from connecting to my_host_IP:trillium_port, anyone on my host could still get to it without authenticating via localhost:trilium_port.

Not very likely and my notes are encrypted, but not perfect either.

**Comment by d03j** [comment]
> > `/api/sync/*` and `/api/login/sync`
> 
> Exceptions for `/api/sync/*`, `/api/login/sync` and `/api/setup/status` worked for me.
> 
> I would still love to have the option to enable Basic Auth on the webserver and API though. :)

Did something change with the last upgrade? This stopped working for me..

**Comment by foozzi** [comment]
> > > `/api/sync/*` and `/api/login/sync`
> > 
> > 
> > Exceptions for `/api/sync/*`, `/api/login/sync` and `/api/setup/status` worked for me.
> > I would still love to have the option to enable Basic Auth on the webserver and API though. :)
> 
> Did something change with the last upgrade? This stopped working for me..

https://github.com/zadam/trilium/issues/4474#issuecomment-1884482155 - works fine

**Comment by meichthys** [comment]
[Trilium has entered maintenance mode](https://github.com/zadam/trilium/issues/4620#issuecomment-2094495698). Future enhancements will be addressed in TrilumNext: https://github.com/TriliumNext/trilium/issues/4968

**Comment by eliandoran** [comment]
This needs to be done at reverse proxy level, if desired.
